src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/RegistrationDataEntry.epics.js

```diff
@@ -41,7 +41,7 @@ const standardGeoJson = (geometry) => {
 const deriveAttributesFromFormValues = (formValues = {}) =>
     Object.keys(formValues)
         .filter(key => !geometryType(key))
-        .map(key => ({ attribute: key, value: formValues[key] }));
+        .map(key => ({ attribute: key, value: typeof formValues[key] === 'object' ? formValues[key]?.id : formValues[key] }));
```

Comment:
I am not sure if this check is enough. Maybe is better to create a convertor and go one by one for each `dataElementTypes`. WDYT?

Reply:
@simonadomnisoru  that makes sense, not all `dataElementTypes` are handled the same